Assets/Code/CameraController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraController : MonoBehaviour
+{
+    public Transform[] viewPoints;
+    public float transformSpeed = 1.0f;
+    private Transform currentView;
+    private int currentIndex = 0;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentView = viewPoints[currentIndex];
+    }
+
+    void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.F))
+        {
+            if(currentIndex == 0)
+            {
+                currentIndex = 1;
+            }else if(currentIndex == 1)
+            {
+                currentIndex = 0;
+            }
+            currentView = viewPoints[currentIndex];
+        }
+        
+    }
+
+    void LateUpdate()
+    {
+        //Lerp position
+        transform.position = Vector3.Lerp(transform.position, currentView.position, Time.deltaTime * transformSpeed);
+
+        Vector3 currentAngle = new Vector3(
+            Mathf.LerpAngle(transform.rotation.eulerAngles.x, currentView.transform.rotation.eulerAngles.x, Time.deltaTime * transformSpeed),
+            Mathf.LerpAngle(transform.rotation.eulerAngles.y, currentView.transform.rotation.eulerAngles.y, Time.deltaTime * transformSpeed),
+            Mathf.LerpAngle(transform.rotation.eulerAngles.z, currentView.transform.rotation.eulerAngles.z, Time.deltaTime * transformSpeed)
```

Comment:
repeated code, please store `transform.rotation.eulerAngles` etc in a temporary variable

eg

```cs
float transformAngles = transform.rotation.eulerAngles;
float currentViewAngles = currentView.transform.rotation.eulerAngles;
float speed = Time.deltaTime * transformSpeed;
// ...
Mathf.LerpAngle(transformAngles.x, currentViewAngles.x, speed),
Mathf.LerpAngle(transformAngles.y, currentViewAngles.y, speed),
Mathf.LerpAngle(transformAngles.z, currentViewAngles.z, speed)
```

etc.

Reply:
i agree lol but again it was like midnight when i was doing this. ill refactor no problem :)

Reply:
np, just ignore this then this was made under the impression that you were more or less done